util_windows.go

```diff
@@ -93,10 +93,30 @@ func readFileOnce(filename string) ([]byte, error) {
 		// cheaply. If the size was wrong, we'll either waste some space off the end
 		// or reallocate as needed, but in the overwhelmingly common case we'll get
 		// it just right.
-		if size := fi.Size() + bytes.MinRead; size > n {
-			n = size
+		n += fi.Size()
+	}
+
+	data := make([]byte, n)
+	remaining := data
+	var total int64
+	for {
+		n, err := f.Read(remaining)
+		remaining = remaining[n:]
+		total += int64(n)
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, err
+		}
+
+		// Size hint undershoot, let's grow the buffer
+		if len(remaining) == 0 {
+			dataSize := len(data)
+			data = append(data, 0) // use append's growth pattern
```

Comment:
This will read one byte at a time unless you update the capacity.

Anyways, I'd just explicitly expand by 2x and make it obvious.